assets/app/components/site/editor/editorContainer.js

```diff
@@ -106,7 +133,63 @@ class Editor extends React.Component {
       path = `${path}.md`;
     }
 
-    siteActions.createCommit(site, path, content, message, sha);
+    return { content, path, message };
+  }
+
+  commitOrPR() {
+    const { site } = this.props;
+    const { path } = this.state;
+    const draftBranch = getDraft(path, site.branches);
+
+    if (!draftBranch) {
+      this.submitFile();
+    }
+
+    this.submitFile(draftBranch.name).then(() => {
```

Comment:
can we come up with a good name for this block and call it instead of embedding the code within this function? We're named "commitOrPR", so I'm guessing the previous `submitFIle()` call is the commit path while the rest of this is the PR path. That's what I get when reading the following lines. I'd like to avoid reading and guessing because I'm very lazy and can only follow so much logic-- I'm happier when you tell me what the abstraction is called so I can move on to what I'm really looking for.